### PR TITLE
Issue #2572 Cover more scenario in registry-route feature file

### LINKED ISSUE #2572: Cover more scenario in addon-registry-route.feature file
### General information
Split the 5th scenario in addon-registry-route.feature file to avail fixes from both iso and addon side.

  * Minishift version: master
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

 
### Steps to reproduce

  1. ```make integrtion GODOG_OPTS="-tags addons-registry-route"```
  2. 
  3. 
  4. 

### Expected

Split [this scenario](https://github.com/minishift/minishift/blob/master/test/integration/features/addons/addon-registry-route.feature#L46) into two as noted down 
```
//iso side fix
Scenario: As a user, I can login to OpenShift integrated docker registry after restart minishift
    Given Minishift has state "Running"
     When executing "minishift stop" succeeds
     Then executing "minishift start" succeeds
      And Minishift should have state "Running"
     Then with up to "60" retries with wait period of "3" seconds container name "registry_docker-registry" should be "running"
      And with up to "60" retries with wait period of "1" second command "curl -ik https://docker-registry-default.$(minishift ip).nip.io" output should contain "200 OK"
      And executing "docker login -u developer -p `oc whoami -t` docker-registry-default.$(minishift ip).nip.io" in host shell succeeds
      And stdout of host shell should contain "Login Succeeded"
```

```
//addon side fix
Scenario: As a user, I can reapply registry-route addon and login to OpenShift integrated docker registry
    Given Minishift has state "Running"
     When executing "minishift addon apply registry-route" succeeds
     Then with up to "60" retries with wait period of "3" seconds container name "registry_docker-registry" should be "running"
      And with up to "60" retries with wait period of "1" second command "curl -ik https://docker-registry-default.$(minishift ip).nip.io" output should contain "200 OK"
      And executing "docker login -u developer -p `oc whoami -t` docker-registry-default.$(minishift ip).nip.io" in host shell succeeds
      And stdout of host shell should contain "Login Succeeded"
```
### Actual
Only addons side fix is added in the feature file. No scenario for iso side fix
### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

### PR DESCRIPTION
**Pull request by amitkrout**
Fixes #2572

Steps to test the pull request

  1. make integration GODOG_OPTS="-tags addon-registry-route"
  2. 
  3. 
  4.

### CODE REVIEW

File: test/integration/features/addons/addon-registry-route.feature [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@addon-registry-route
+@addon-registry-route @core
```
Comment (amitkrout) [inline test/integration/features/addons/addon-registry-route.feature]: Added ```@core``` Just for testing in ci

**Comment by amitkrout** [comment]
@praveenkumar I have split up the scenario to cover registry login scenario after minishift restart. 

Now the question is do we really need reapplying of registry route addon ? Because registry route login is successful even after restart (enable registry route -> m start -> m stop -> start -> registry login).

**Comment by amitkrout** [comment]
retest this please

**Comment by praveenkumar** [comment]
> Now the question is do we really need reapplying of registry route addon ?

@amitkrout yes because we want our registry addon to be as atomic as possible so we know that reapply doesn't break anything.

**Comment by amitkrout** [comment]
retest this please

**Comment by agajdosi** [comment]
@amitkrout PR #2869 has been merged, you can go on this PR once you are back :)

**Comment by amitkrout** [comment]
retest this please

File: test/integration/features/addons/addon-registry-route.feature [hunk 2]
```diff
@@ -57,17 +57,22 @@ so that user can able to push their container images directly to registry and de
      Then with up to "10" retries with wait period of "500ms" the "status code" of HTTP request to "/" of service "nginx" in namespace "myproject" is equal to "200"
       And with up to "10" retries with wait period of "500ms" the "body" of HTTP request to "/" of service "nginx" in namespace "myproject" contains "Welcome to nginx!"
 
-  Scenario: User restarts Minishift
+  Scenario: User can login to OpenShift integrated docker registry after Minishift restart
     Given Minishift has state "Running"
      When executing "minishift stop" succeeds
      Then executing "minishift start" succeeds
+      And Minishift should have state "Running"
+     Then with up to "20" retries with wait period of "3" seconds container name "registry_docker-registry" should be "running"
```
Comment (agajdosi) [inline test/integration/features/addons/addon-registry-route.feature]: Please put this and following steps into a separate scenario. Scenario: Restart, Scenario: User can login to OpenShift integrated docker registry

File: test/integration/features/addons/addon-registry-route.feature [hunk 3]
```diff
@@ -57,17 +57,22 @@ so that user can able to push their container images directly to registry and de
...
+      And with up to "60" retries with wait period of "1000ms" the "status code" of HTTP request to "/" of service "docker-registry" in namespace "default" is equal to "200"
+     When executing "docker login -u developer -p `oc whoami -t` docker-registry-default.$(minishift ip).nip.io" in host shell succeeds
+     Then stdout of host shell should contain "Login Succeeded"
 
-  Scenario: User can login to OpenShift integrated docker registry after restart minishift
+  Scenario: User can reapply registry-route addon and login to OpenShift integrated docker registry
```
Comment (agajdosi) [inline test/integration/features/addons/addon-registry-route.feature]: This also can be split into 2 scenarios. Scenario: User can reapply addon and Scenario: Login to OpenShift registry

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
:+1: for https://github.com/minishift/minishift/issues/2907

File: test/integration/features/addons/addon-registry-route.feature (hunk 3 above)
Comment (amitkrout) [reply test/integration/features/addons/addon-registry-route.feature]: sure

**Comment by praveenkumar** [comment]
@agajdosi so we should merge this and there is already bug for push to registry failure.

**Comment by amitkrout** [comment]
@praveenkumar @agajdosi This can be mergeable. If you see the changes made to this pr that does not depend on the registry psuh.

**Comment by agajdosi** [comment]
@praveenkumar Seems like we could merge this.

**Pull request merged**